Woof/Woof/Model/Preferences/PreferencesHandler.swift

```diff
@@ -0,0 +1,59 @@
+import Foundation
+
+/**
+ A type used for managing user preferences in the application.
+ */
+enum PreferencesHandler {
+    // MARK: - Internal interface
+
+    /**
+     Saves the selected role as a user preference.
+
+     - Parameters:
+        - userRole: The selected role to be saved.
+
+     - Returns: A boolean value indicating whether the saving operation was successful.
+     */
+    @discardableResult static func set(userRole: Role) -> Bool {
+        var preferences: Preferences = loadPreferences() ?? Preferences()
+
+        preferences.selectedRole = userRole
+
+        return store(preferences)
+    }
+
+    /**
+     Loads the selected role from user preferences.
```

Comment:
What is a "selected" role?

Reply:
`current` is only misleading
fixed

Reply:
Ok, you have updated the `getUserRole()` method. But the `set(userRole)` method contains information about the "selected" role again...

Does it too hard to double-check the PR before the re-request? ;) 

Also would suggest to replace the:
> Loads the current role from user preferences

with:

> Gets the current user role in the app.

to show what do you mean by "role" and avoid duplicating (we know already that the `PreferencesHandler` loads user preferences, don't we?)

Reply:
Yes, I agree with your suggestion about avoiding duplicating.
fixed

Reply:
Yes, I agree with your suggestion about avoiding duplicating.
fixed